.github/workflows/linux_conditional_compilation.yml

```diff
@@ -0,0 +1,134 @@
+      OPENVINO_CONTRIB_REPO: ${{ github.workspace }}/openvino_contrib
+      BUILD_DIR: ${{ github.workspace }}/build
+      DATA_PATH: ${{ github.workspace }}/testdata
+      MODELS_PATH: ${{ github.workspace }}/testdata
+      OV_TEMP: ${{ github.workspace }}/openvino_temp
+      PYTHON_STATIC_ARGS: -m "not dynamic_library and not template_plugin"
+    steps:
+      - name: Clone OpenVINO
+        uses: actions/checkout@v3
+        with:
+          path: 'openvino'
+          submodules: 'recursive'
+
+      - name: Clone testdata for C API tests
+        uses: actions/checkout@v3
+        with:
+          repository: 'openvinotoolkit/testdata'
+          path: 'testdata'
+          submodules: 'recursive'
+          lfs: 'true'
+
+      #
+      # Dependencies
+      #
+
+      - name: Install build dependencies
+        run: |
+          sudo -E ${{ env.OPENVINO_REPO }}/install_build_dependencies.sh
+          sudo -E apt update
+          sudo -E apt --assume-yes install openjdk-11-jdk libbz2-dev clang unzip libpugixml-dev libtbb-dev intel-opencl-icd ocl-icd-opencl-dev opencl-headers
+
+          wget https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip
+          unzip ninja-linux.zip
+          sudo cp -v ninja /usr/local/bin/
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      #
+      # Build
+      #
+
+      - name: Get tools versions
+        run: |
+          ninja --version || exit 1
+          python3 --version || exit 1
+          cmake --version || exit 1
+
+      - name: Get number of CPU cores
+        uses: SimenB/github-actions-cpu-cores@v1
+        id: cpu-cores
+
+      - name: CMake configure CC COLLECT
+        run: |
+          cmake \
+            -G "Ninja Multi-Config" \
+            -DENABLE_CPPLINT=OFF \
+            -DENABLE_GAPI_PREPROCESSING=OFF \
+            -DCMAKE_VERBOSE_MAKEFILE=ON \
+            -DCMAKE_COMPILE_WARNING_AS_ERROR=OFF \
+            -DENABLE_FASTER_BUILD=ON \
+            -DENABLE_PROFILING_ITT=ON \
+            -DSELECTIVE_BUILD=COLLECT \
+            -S ${{ env.OPENVINO_REPO }} \
+            -B ${{ env.BUILD_DIR }}
+
+      - name: Clean ccache stats
+        run: ccache --zero-stats --show-config
+
+      - name: Build CC COLLECT
+        run: cmake --build ${{ env.BUILD_DIR }} --parallel ${{ steps.cpu-cores.outputs.count }} --config Release --target openvino_intel_cpu_plugin openvino_ir_frontend benchmark_app sea_itt_lib
+
+      - name: Show ccache stats
+        run: ccache --show-stats
+
+      - name: Code usage analysis
+        run: |
+          python3 ${{ env.OPENVINO_REPO }}/thirdparty/itt_collector/runtool/sea_runtool.py \
+            --bindir ${{ env.OPENVINO_REPO }}/bin/intel64/Release -o ${{ env.BUILD_DIR }}/itt_stat ! \
+          ${{ env.OPENVINO_REPO }}/bin/intel64/Release/benchmark_app -niter 1 -nireq 1 \
+            -m ${{ env.MODELS_PATH }}/models/test_model/test_model_fp32.xml -d CPU
```

Comment:
I recommend to use several real models instead of dummy test_modelt
We can download models from public model hubs (TF, others) and use it here.

Reply:
Is there any way we could download such model(s) w/o building and installing the Python wheels for model download and convertation? Download IRs directly from some storage?

Or do you suggest extending the workflow beyond what is in Azure with Python setup, dependencies setup, wheels building, and model downloading and convertation?

Reply:
@ilya-lavrenov , I would suggest we move the pipelines without modifications first, and improve after it's working.